multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ObjectStoreFileStorage.java

```diff
@@ -44,17 +47,19 @@ public ObjectStoreFileStorage(BlobStore blobStore, String container) {
     }
 
     @Override
-    public void addFile(FileEntry fileEntry, InputStream content) throws FileStorageException {
+    public void addFile(FileEntry fileEntry, InputStream content) throws FileStorageException, IOException {
         String entryName = fileEntry.getId();
         long fileSize = fileEntry.getSize()
                                  .longValue();
+        LOGGER.trace("Building blob ...");
         Blob blob = blobStore.blobBuilder(entryName)
-                             .payload(content)
+                             .payload(ByteSource.wrap(ByteStreams.toByteArray(content)))
```

Comment:
This will read the entire input stream in memory - [source code](https://github.com/google/guava/blob/master/guava/src/com/google/common/io/ByteStreams.java#L173)
This wouldn't be a problem for small MTARs but it does not scale.
The issue is somewhat mitigated by the small usage of the _deploy from URL_ feature but is still worth considering.

If we are going to read the whole file in memory, prefer to use the standard library or apache commons (IOUtils.toByteArray) so we wouldn't need to include the Guava library as a dependency (it is quite sizable and will only bloat our executable)